src/tools/dashboard/src/reference.rs

```diff
@@ -151,72 +155,104 @@ fn extract(par_from: &Path, par_to: &Path) -> Vec<(Example, PathBuf)> {
         let line = BufReader::new(file).lines().nth(from.line - 1).unwrap().unwrap();
-        if line.contains("edition2015") {
-            prepend_text(to, "// compile-flags: --edition 2015");
-        } else {
-            prepend_text(to, "// compile-flags: --edition 2018");
+        if !line.contains("edition2015") {
+            props.rustc_args.push(String::from("--edition"));
+            props.rustc_args.push(String::from("2018"));
         }
-        // Most examples with `compile_fail` configuration fail because of
-        // check errors.
-        if line.contains("compile_fail") {
-            prepend_text(to, "// rmc-check-fail");
+        // Most examples with `compile_fail` annotation fail because of check
+        // errors. This heuristic can be overridden by manually specifying the
+        // fail step in the corresponding config file.
+        if props.fail_step.is_none() && line.contains("compile_fail") {
+            props.fail_step = Some(FailStep::Check);
         }
         // RMC should catch run-time errors.
-        if line.contains("should_panic") {
-            prepend_text(to, "// rmc-verify-fail");
+        if props.fail_step.is_none() && line.contains("should_panic") {
+            props.fail_step = Some(FailStep::Verification);
         }
+        // Prepend those properties to test/example file.
+        prepend_props(&props);
     }
-    // For now, we will only manually pre-process the tests that cause infinite loops.
-    // TODO: Add support for manually adding options and assertions (see issue #324).
-    let loop_tests: [PathBuf; 4] = [
-        ["src", "test", "ref", "Appendices", "Glossary", "263.rs"].iter().collect(),
-        ["src", "test", "ref", "Linkage", "190.rs"].iter().collect(),
-        [
-            "src",
-            "test",
-            "ref",
-            "Statements and expressions",
-            "Expressions",
-            "Loop expressions",
-            "133.rs",
-        ]
-        .iter()
-        .collect(),
-        [
-            "src",
-            "test",
-            "ref",
-            "Statements and expressions",
-            "Expressions",
-            "Method call expressions",
-            "10.rs",
-        ]
-        .iter()
-        .collect(),
-    ];
-
-    for test in loop_tests {
-        let code = fs::read_to_string(&test).unwrap();
-        let code = format!("// rmc-flags: --cbmc-args --unwind 1\n{}", code);
-        fs::write(&test, code).unwrap();
+    if !config_paths.is_empty() {
+        panic!(
+            "Error: The examples corresponding to the following config files \
+             were not encountered in the pre-processing step:\n{}This is most \
+             likely because the line numbers of the config files are not in \
+             sync with the line numbers of the corresponding code blocks in \
+             the latest versions of the Rust books. Please update the line \
+             numbers of the config files and rerun the program.",
```

Comment:
Have you chatted with the group about this? Sounds like this approach might be quite flaky?

I don't have any alternative ideas off the top of my head, though. Other than trying to find some git-fu that could automatically update the line numbers somehow.

Reply:
Yes. We talked about this in a previous Deep Dive. We considered automatically updating the line number, but that can introduce silent success/failure if the code-blocks change too. Nathan approved of this as a temporary solution for now, though it adds to the list of problems Adrian has to deal with in his weekly rebases.